lib/avro_turf.rb

```
@@ -40,11 +41,30 @@ def decode(encoded_data, schema_name: nil)
     schema = schema_name && resolve_schema(schema_name)
     reader = Avro::IO::DatumReader.new(nil, schema)
     dr = Avro::DataFile::Reader.new(io, reader)
+
+    # If no reader schema is specified, we'll try to find one matching the name
+    # of the writer's schema.
+    if schema_name.nil?
+      schema = load_readers_schema(reader.writers_schema)
+      reader.readers_schema = schema if schema
+    end
+
     dr.first
   end
 
   private
 
+  # Returns the reader schema matching the writer's schema if one is locally
+  # available, or nil if there's no matching reader schema.
+  def load_readers_schema(writers_schema)
+    if writers_schema.respond_to?(:fullname)
+      schema_name = writers_schema.fullname
+      resolve_schema(schema_name)
+    end
+  rescue SchemaNotFoundError
+    nil
```

Lets at least put a logger statement here. Don't like eating errors like this.